src/v/cluster/partition_manager.h

```diff
@@ -97,14 +100,23 @@ class partition_manager {
         _manage_watchers.unregister_notify(id);
     }
 
+    /*
+     * read-only interface to partitions.
+     *
+     * note that users of this interface must take care not to hold iterators
```

Comment:
I think the node-hash might have integrator stability. 

I don’t see major value in flat-hash here

Reply:
node hash map doesn't have iterator stability (it does have pointer stability on the values which may be what you're thinking)

Reply:
Ah good note.